prometheus_hardware_exporter/collector.py

```diff
@@ -547,13 +548,16 @@ def fetch(self) -> List[Payload]:
                     "Unknown ipmi SEL state: %s. Treating it as Nominal.", sel_entry["State"]
                 )
                 sel_state_value = sel_states_dict["NOMINAL"]
+
+            key = (sel_entry["Name"], sel_entry["Type"])
+            if key not in sel_entries_dict or sel_entries_dict[key] < sel_state_value:
```

Comment:
**nitpick:** 

I think we miss some tests to verify the merging logic.

Reply:
All of the merging conditions are tested since the [sample data](https://github.com/sudeephb/prometheus-hardware-exporter/blob/main/tests/unit/test_resources/ipmi/ipmi_sample_data.py#L1) contains all of the possible conditions(repeated data,  repeated data with only the `state` difference).

Reply:
How do I know that the value will be `Warning` instead of `Nominal` if there are two records?
The simple way is we check the return payload's details.